data-management/data-interface.js

```diff
@@ -12,7 +12,7 @@ async function execute(fn) {
     try {

```

Comment:
If we change this to throw the error rather than return it, then we can remove the entire try catch block and any errors will be thrown by default